### PR TITLE
Overload getResourcesByType method with tenantId

### PR DESCRIPTION
**Pull request by PasinduYeshan**
### Proposed changes in this pull request
$subject
 The `getResourcesByType` method has been overloaded to facilitate the retrieval of resources based on the provided tenantId.

### Related Issues
- https://github.com/wso2/product-is/issues/18247

### CODE REVIEW

File: components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManager.java [hunk 1]
```diff
@@ -100,6 +100,19 @@ Resources getTenantResources(String tenantDomain, Condition searchCondition) thr
      */
     Resources getResources() throws ConfigurationManagementException;
 
+    /**
+     * Get all the resources belongs to the given {@link ResourceType} for a given tenant ID.
```
Comment (dewniMW) [inline components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManager.java:104]:
```suggestion
     * Get all the resources belonging to the given {@link ResourceType} for a given tenant ID.
```

File: components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManager.java [hunk 2]
```diff
@@ -100,6 +100,19 @@ Resources getTenantResources(String tenantDomain, Condition searchCondition) thr
...
+
+     * @param tenantId         The ID of the tenant.
+     * @param resourceTypeName {@link ResourceType} object name.
+     * @return 200 ok. @link Resources} object with all the resources of the given resource type name.
```
Comment (dewniMW) [inline components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManager.java:108]: 200 ok?

Comment (PasinduYeshan) [reply components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManager.java:108]: Yes I used the same comment as the overloaded method.

Comment (PasinduYeshan) [reply components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManager.java]: Removed

File: components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManager.java (hunk 1 above)
Comment (PasinduYeshan) [reply components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManager.java]: Changed

File: components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManager.java [hunk 3]
```diff
@@ -101,10 +101,23 @@ Resources getTenantResources(String tenantDomain, Condition searchCondition) thr
...
+     * Get all the resources belonging to the given {@link ResourceType} for a given tenant ID.
+
+     * @param tenantId         The ID of the tenant.
+     * @param resourceTypeName {@link ResourceType} object name.
+     * @return {@link Resources} object with all the resources of the given resource type name.
+     * @throws ConfigurationManagementException Configuration Management Exception.
+     */
+   default Resources getResourcesByType(int tenantId, String resourceTypeName) throws ConfigurationManagementException {
+
+            throw new NotImplementedException("This functionality is not implemented.");
```
Comment (dewniMW) [inline components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManager.java:113]: is this formatted correctly?

Comment (PasinduYeshan) [reply components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManager.java]: Reformatted

**Review by dewniMW — APPROVED** [review]

**Pull request merged**